upgrades/environ.go

```diff
@@ -55,20 +53,24 @@ func (e environConfigGetter) CloudSpec(names.ModelTag) (environs.CloudSpec, erro
 	return e.m.CloudSpec()
 }
 
-func newEnvironUpgradeOperation(op environs.UpgradeOperation) Operation {
+func newEnvironUpgradeOperation(op environs.UpgradeOperation, controllerUUID string) Operation {
 	steps := make([]Step, len(op.Steps))
 	for i, step := range op.Steps {
-		steps[i] = newEnvironUpgradeStep(step)
+		steps[i] = newEnvironUpgradeStep(step, controllerUUID)
 	}
-	return upgradeToVersion{op.TargetVersion, steps}
+	// NOTE(axw) all two of the current environ upgrade steps will happily
+	// run idempotently; there are no post-upgrade steps that will render
+	// them non-runabble. This is here as a backstop, to ensure the upgrades
```

Comment:
typo - runnable.
Also, I don't really understand this comment. Are you referring to the use of `jujuversion.Current` instead of `op.TargetVersion`? Why does that make it a backstop?